src/controllers/controllerpreset.h

```diff
@@ -43,16 +45,38 @@ class ControllerPreset {
      * @param filename Name of the XML file to add
      * @param functionprefix Function prefix to add
      */
-    void addScriptFile(QString filename, QString functionprefix,
-                       bool builtin=false) {
+    void addScriptFile(QString filename, QString functionprefix, bool builtin = false) {
         ScriptFileInfo info;
         info.name = filename;
         info.functionPrefix = functionprefix;
+        // Always try to load script from the mapping's directory first
+        info.file = QFileInfo(dirPath().absoluteFilePath(filename));
         info.builtin = builtin;
-        scripts.append(info);
+        m_scripts.append(info);
         setDirty(true);
     }
 
+    QList<ScriptFileInfo> getScriptFiles(const QString& fallbackPath = QString()) const {
+        if (fallbackPath.isEmpty()) {
+            return m_scripts;
+        }
+
+        QDir path(fallbackPath);
+        QList<ScriptFileInfo> scriptsWithFallbackPath;
+        for (const ScriptFileInfo& script : m_scripts) {
+            qDebug() << "script" << script.file;
+            if (script.file.exists()) {
+                scriptsWithFallbackPath << script;
```

Comment:
The variable name `scriptsWithFallbackPath` is confusing here. At this line, it is scripts without fallback paths getting added. How about renaming the QList to `scriptsIncludingFallbacks`?